wrapper/src/main/java/software/amazon/jdbc/plugin/efm/MonitorServiceImpl.java

```diff
@@ -100,16 +100,14 @@ public MonitorConnectionContext startMonitoring(
           new Object[] {hostSpec}));
     }
 
-    final Monitor monitor;
-    if (this.cachedMonitor == null
+    Monitor monitor = this.cachedMonitor == null ? null : this.cachedMonitor.get();
```

Comment:
```suggestion
    final Monitor monitor = this.cachedMonitor == null ? null : this.cachedMonitor.get();
```

Reply:
Can't do that. Monitor maybe reassigned later (line 109).